### PR TITLE
Send end frame when filtering for only historical messages

### PR DESCRIPTION
**Pull request by akrambek**

### CODE REVIEW

File: src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheClientFetchFactory.java [hunk 1]
```diff
@@ -243,6 +244,8 @@ public MessageConsumer newStream(
             }
 
             final KafkaFilterCondition condition = cursorFactory.asCondition(filters);
+            final KafkaAge maximumAge = filters.anyMatch(a -> a.conditions().anyMatch(c -> c.age().get() != KafkaAge.HISTORICAL)) ?
+                KafkaAge.LIVE : KafkaAge.HISTORICAL;
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheClientFetchFactory.java:248]: Let's use static imports for `LIVE` and `HISTORICAL`.

File: src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheClientFetchFactory.java [hunk 2]
```diff
@@ -243,6 +244,8 @@ public MessageConsumer newStream(
             }
 
             final KafkaFilterCondition condition = cursorFactory.asCondition(filters);
+            final KafkaAge maximumAge = filters.anyMatch(a -> a.conditions().anyMatch(c -> c.age().get() != KafkaAge.HISTORICAL)) ?
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheClientFetchFactory.java:247]: Do we need to check `c.kind() == AGE` before calling `c.age()` in the matcher?
Also we can probably hoist this stateless lambda into a member variable of the factory.

File: src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheClientFetchFactory.java [hunk 3]
```diff
@@ -956,6 +963,13 @@ private void doClientReplyDataIfNecessary(
                 {
                     break;
                 }
+
+                if (maximumAge == KafkaAge.HISTORICAL && cursor.offset > initialGroupLatestOffset)
```
Comment (jfallows) [inline src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheClientFetchFactory.java:967]: Let's use static import for `HISTORICAL`.

Does `>` condition work even if there are no live messages after `initialGroupLatestOffset`?

Comment (akrambek) [reply src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheClientFetchFactory.java]: Yep that works

**Pull request merged**